go/pkg/utils/detector_test.go

```diff
@@ -0,0 +1,1108 @@
+		"f1.file",
+		"f2.txt",
+	}
+
+	for _, path := range testFiles {
+		err := os.MkdirAll(filepath.Join(tempDir, filepath.Dir(path)), 0755)
+		if err != nil {
+			t.Errorf("Failed to create test dir: %v", err)
+		}
+		_, err = os.Create(filepath.Join(tempDir, path))
+		if err != nil {
+			t.Errorf("Failed to create test file: %v", err)
+		}
+	}
+
+	tests := []struct {
+		name           string
+		root           string
+		propsFiles     []model.ApplicationFileInfo
+		expectedResult string
+	}{
+		{
+			name: "Case 1: Matching file found with regex",
+			root: tempDir,
+			propsFiles: []model.ApplicationFileInfo{
+				{File: ".*.txt", Dir: ""},
+			},
+			expectedResult: filepath.Join(tempDir, "f2.txt"),
+		},
+		{
+			name: "Case 2: Matching file found exact",
+			root: tempDir,
+			propsFiles: []model.ApplicationFileInfo{
+				{File: "f1.file", Dir: ""},
+			},
+			expectedResult: filepath.Join(tempDir, "f1.file"),
+		},
+		{
+			name: "Case 3: No matching file found",
+			root: tempDir,
+			propsFiles: []model.ApplicationFileInfo{
+				{File: "missing.file", Dir: "."},
+			},
+			expectedResult: "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetAnyApplicationFilePath(tt.root, tt.propsFiles, &ctx)
+			if result != tt.expectedResult {
+				t.Errorf("Expected result %s, got %s", tt.expectedResult, result)
+			}
+		})
+	}
+}
+
+func TestGetAnyApplicationFilePathExactMatch(t *testing.T) {
+	tempDir := t.TempDir()
+
+	testFiles := []string{
+		"f1.txt",
+		"subdir/f1.txt",
+	}
+
+	for _, path := range testFiles {
+		err := os.MkdirAll(filepath.Join(tempDir, filepath.Dir(path)), 0755)
+		if err != nil {
+			t.Errorf("Failed to create test dir: %v", err)
+		}
+		_, err = os.Create(filepath.Join(tempDir, path))
+		if err != nil {
+			t.Errorf("Failed to create test file: %v", err)
+		}
+	}
+
+	tests := []struct {
+		name           string
+		root           string
+		propsFiles     []model.ApplicationFileInfo
+		expectedResult string
+	}{
+		{
+			name: "Case 1: Matching file found in nested dir",
+			root: tempDir,
+			propsFiles: []model.ApplicationFileInfo{
+				{File: "f1.txt", Dir: "subdir"},
+				{File: "f1.txt", Dir: ""},
+			},
+			expectedResult: filepath.Join(tempDir, "subdir/f1.txt"),
+		},
+		{
+			name: "Case 2: Matching file found in root",
+			root: tempDir,
+			propsFiles: []model.ApplicationFileInfo{
+				{File: "f1.txt", Dir: ""},
+				{File: "f1.txt", Dir: "subdir"},
+			},
+			expectedResult: filepath.Join(tempDir, "f1.txt"),
+		},
+		{
+			name: "Case 3: No matching file found",
+			root: tempDir,
+			propsFiles: []model.ApplicationFileInfo{
+				{File: ".*.txt", Dir: ""},
+			},
+			expectedResult: "",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetAnyApplicationFilePathExactMatch(tt.root, tt.propsFiles)
+			if result != tt.expectedResult {
+				t.Errorf("Expected result %s, got %s", tt.expectedResult, result)
+			}
+		})
+	}
+}
+
```

Comment:
Shouldn't we add a test for `readAnyApplicationFile` too?

Reply:
Will add in next commit along with cache.go and language_file_handler.go tests 👍